starknet_py/utils/typed_data/functions.py

```diff
@@ -0,0 +1,89 @@
+from typing import List, Union
+
+from starkware.cairo.common.hash_state import compute_hash_on_elements
+from starkware.starknet.public.abi import get_selector_from_name
+
+from starknet_py.cairo.felt import encode_shortstring
+from starknet_py.utils.typed_data.types import TypedData
+
+
+def get_hex(value: Union[int, str]):
+    if isinstance(value, str):
+        if value[:2] == "0x":
+            return value
+        return hex(encode_shortstring(value))
+    return hex(value)
+
+
+def get_dependencies(typed_data: TypedData, type_: str) -> List[str]:
+    if type_ and type_[-1] == "*":
+        type_ = type_[:-1]
+
+    if type_ not in typed_data.types:
+        return []
+    dependencies = set()
+
+    def get_deps(type_: str):
+        for item in typed_data.types[type_]:
+            item_type = item["type"]
+            if item_type in typed_data.types and item_type not in dependencies:
+                dependencies.add(item_type)
+                get_deps(item_type)
+
+    get_deps(type_)
+    return [type_, *list(dependencies)]
+
+
+def encode_type(typed_data: TypedData, type_: str) -> str:
```

Comment:
This should accept some better representation of Struct than the entire TypedData. In shouldn't be the concern of this method to also parse the data.